### PR TITLE
Whisper Export

### PR DESCRIPTION
**Pull request by petermcaughan**
### Description
Add scripts to export Whisper model to ONNX and integrate the ORT BeamSearch op with the resulting graphs.

Example command to execute this script:

python convert_to_onnx.py -m openai/whisper-large --output whisper -e

### CODE REVIEW

File: onnxruntime/python/tools/transformers/models/whisper/whisper_helper.py [hunk 1]
```diff
@@ -0,0 +1,269 @@
...
+
+PRETRAINED_WHISPER_MODELS = ["whisper-tiny", "whisper-tiny.en",
+                        "whisper-small", "whisper-small.en",
+                        "whisper-medium", "whisper-medium.en",
+                        "whisper-base", "whisper-base.en",
+                        "whisper-large", "whisper-large-v2",
+                        ]
+
+
+class WhisperHelper:
```
Comment (justinchuby) [inline onnxruntime/python/tools/transformers/models/whisper/whisper_helper.py:36]: Also avoid names like "helper," where anything can go in

File: onnxruntime/python/tools/transformers/models/whisper/past_helper.py
Comment (justinchuby) [inline onnxruntime/python/tools/transformers/models/whisper/past_helper.py:1]: I would consider a more meaningful name than "helper"

**Comment by justinchuby** [comment]
Hi! Just flying by and left some nits. For locally running lintrunner (and running autofix), I would refer to https://github.com/microsoft/onnxruntime/blob/main/.lintrunner.toml for usage.

File: onnxruntime/python/tools/transformers/models/whisper/whisper_helper.py [hunk 2]
```diff
@@ -0,0 +1,276 @@
...
+        if is_float16:
+            optimization_options = FusionOptions("bart")
+            optimization_options.enable_skip_layer_norm = False
+
+        if (use_multihead_attn_fusion):
+            if (optimization_options is None):
+                optimization_options = FusionOptions("bart")
+            optimization_options.use_multi_head_attention = True
+
+        pdb.set_trace
```
Comment (stevenlix) [inline onnxruntime/python/tools/transformers/models/whisper/whisper_helper.py:241]: Do you need trace here? Either remove the line or import pdb.

**Review by yufenglee — DISMISSED** [review]

**Review by yufenglee — APPROVED** [review]

**Pull request merged**